### PR TITLE
Check node version when using the generator. Need at least 6.9.0+

### PR DESCRIPTION
**Pull request by pascalgrimaud**
Related to https://github.com/jhipster/generator-jhipster/issues/5065
I don't put `fix` or `close` as @gmarziou suggested another solution too

```
Welcome to the JHipster Generator v4.0.0
Documentation for creating an application: https://jhipster.github.io/creating-an-app/
Application files will be generated in folder: /home/pgrimaud/tmp/23-toto
WARNING! Your node version is too old (v4.5.0). You should use at least Node v6.9.0+
? (1/15) Which *type* of application would you like to create? (Use arrow keys)
❯ Monolithic application (recommended for simple projects) 
  Microservice application 
  Microservice gateway 
  [BETA] JHipster UAA server (for microservice OAuth2 authentication) 
```

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
After reading the stackoverflow post, this solution is more permissive (a warning) and more consistent with other checks version we already have, so I think we can close #5065 ?

**Comment by jdubois** [comment]
Not sure if the warning is enough, as people will probably ignore it, but then that's indeed more consistent.... And who knows, maybe it could work with some older versions...

**Comment by deepu105** [comment]
@pascalgrimaud @jdubois I dont think we should do it this way. This would also require manually maintaining the version in the prompts. The below in package should have worked so I would investigate that first and even if we are doing a check we should check against whats defined as engine instead of a hardcoded check
```
  "engines": {
    "node": ">=6.9.0"
  },
```

**Comment by pascalgrimaud** [comment]
Agree with you @deepu105

**Comment by jdubois** [comment]
Yes agreed @deepu105 - and I really don't get why the "engines" part of the package.json doesn't work.

**Comment by deepu105** [comment]
@jdubois I think it works but it just prints a warning so may be people dont read that warning :)

**Comment by jdubois** [comment]
Oh... If that's the case then I'll be quite angry with all those bug reports.... Anyway it's clearly written in our documentation.

**Comment by deepu105** [comment]
Oh wait but then we are using yarn so we need to check if Yarn prints warnings as well. But NPM warnings are quite noisy so easy for people to miss it actually. They shouldnt have removed the strictEngine keyword

**Comment by jdubois** [comment]
And we got a bug on the new check: https://twitter.com/ecukalla/status/827848436364013568

File: generators/app/index.js [hunk 1]
```diff
@@ -108,6 +108,24 @@ module.exports = JhipsterGenerator.extend({
...
+            if (this.skipChecks || this.skipServer) return;
+            var done = this.async();
+            exec('node -v', function (err, stdout, stderr) {
+                if (err) {
+                    this.warning('Node is not found on your computer.');
+                } else {
+                    var nodeVersionMajor = stdout.split('.')[0].replace(/v/g, '');
+                    var nodeVersionMinor = stdout.split('.')[1];
+                    var nodeVersion = stdout.replace(/\n/g, '');
+                    if (nodeVersionMajor < 6 || (nodeVersionMajor >= 6 && nodeVersionMinor < 9)) {
```
Comment (deepu105) [inline generators/app/index.js]: Btw if we are checking this then better to use the semver library to do that we already use that  in `isJhipsterVersionLessThan` method in `generator-base.js`

Comment (anjia0532) [inline generators/app/index.js]: my node version is v7.5.0 is least version but `yo jhipster` waring `). You should use at least Node v6.9.0+v7.5.0`
**(nodeVersionMajor >= 6 && nodeVersionMinor < 9)** is not ok

**Comment by pascalgrimaud** [comment]
Oh sorry, I coded it too fast... :(

**Comment by pascalgrimaud** [comment]
It should be == instead of >=...

**Comment by anjia0532** [comment]
@pascalgrimaud  yes ,should be ==, my PR https://github.com/jhipster/generator-jhipster/pull/5090

**Pull request merged**